netmap/netmap_contract.go

```diff
@@ -685,9 +717,9 @@ func setConfig(ctx storage.Context, key, val any) {
 }
 
 func cleanup(ctx storage.Context, epoch int) {
-	balanceContractAddr := storage.Get(ctx, balanceContractKey).(interop.Hash160)
-	contract.Call(balanceContractAddr, cleanupEpochMethod, contract.All, epoch)
-
-	containerContractAddr := storage.Get(ctx, containerContractKey).(interop.Hash160)
-	contract.Call(containerContractAddr, cleanupEpochMethod, contract.All, epoch)
+	it := storage.Find(ctx, newEpochSubscribersPrefix, storage.RemovePrefix|storage.KeysOnly)
+	for iterator.Next(it) {
+		contractHash := interop.Hash160(iterator.Value(it).([]byte)[1:]) // one byte is for number prefix
+		contract.Call(contractHash, cleanupEpochMethod, contract.All, epoch)
```

Comment:
And one more important moment that just came to my mind. If something (panic, I mean) happens in the calling callback, then the rest of callbacks won't be executed. It remains `PostPersist` and technically it's OK to fail immediately, so just a question: do we need to think about some exception handlers here?

Reply:
> do we need to think about some exception handlers here?

only our own contracts are expected to be new-epoch-notified so any panic is a global panic to me and i can not say that some contract is more or less important to us, we need all of them to be notified. failed notifications are a failed NeoFS network to me. also, i do know what such a handler should do. notify, panic but later, log?

@roman-khimov

Reply:
Partial failure in many ways is better than complete failure, so handling exceptions makes some sense here, but I'd leave it to another issue.